packages/core/src/external/surescripts/command/convert-batch-response/convert-batch-response-cloud.ts

```
@@ -0,0 +1,38 @@
+import { Config } from "../../../../util/config";
+import { executeWithNetworkRetries } from "@metriport/shared";
+import { getLambdaResultPayload, makeLambdaClient } from "../../../aws/lambda";
+import { SurescriptsConversionBundle, SurescriptsJob } from "../../types";
+import { SurescriptsConvertBatchResponseHandler } from "./convert-batch-response";
+export class SurescriptsConvertBatchResponseHandlerCloud
+  implements SurescriptsConvertBatchResponseHandler
+{
+  constructor(
+    private readonly surescriptsConvertBatchResponseLambdaName: string,
+    private readonly lambdaClient = makeLambdaClient(Config.getAWSRegion())
+  ) {}
+
+  async convertBatchResponse(job: SurescriptsJob): Promise<SurescriptsConversionBundle[]> {
+    const payload = JSON.stringify(job);
+    let resultPayload: string | undefined;
+    await executeWithNetworkRetries(async () => {
+      const result = await this.lambdaClient
+        .invoke({
+          FunctionName: this.surescriptsConvertBatchResponseLambdaName,
+          InvocationType: "RequestResponse",
+          Payload: payload,
+        })
+        .promise();
+
+      resultPayload = getLambdaResultPayload({
+        result,
+        lambdaName: this.surescriptsConvertBatchResponseLambdaName,
+      });
+    });
+
+    if (resultPayload) {
+      const conversionBundles = JSON.parse(resultPayload);
+      return conversionBundles as SurescriptsConversionBundle[];
+    }
+    return [];
```

any reason to not just let it fail?

---

Or rather... what are the cases where `resultPayload` would be undefined?